src/devices/gt_wt_02.c

```diff
@@ -70,7 +70,7 @@ static int gt_wt_02_process_row(r_device *decoder, bitbuffer_t *bitbuffer, int r
 
     // accept only correct checksums, (sum of nibbles modulo 64)
     if ((sum_nibbles & 0x3F) != checksum)
-        return 0; // DECODE_FAIL_MIC
+        return DECODE_FAIL_MIC;
```

Comment:
This will mess up `counter`. Should have been commented better.

Reply:
I will add a > 0 check in the return function.

Reply:
The comment there says "return on first successful", and that's what really happens too.
Guess the logic needs some reworking, saving the last fail code and return on counter > 0 or with the fail code in counter after the loop.